core/src/main/java/org/elasticsearch/index/query/AndQueryBuilder.java

```diff
@@ -110,13 +110,19 @@ public Query toQuery(QueryParseContext parseContext) throws QueryParsingExceptio
         }
 
         BooleanQuery query = new BooleanQuery();
+        int addedFilters = 0;
         for (QueryBuilder f : filters) {
             Query innerQuery = f.toQuery(parseContext);
             // ignore queries that are null
             if (innerQuery != null) {
                 query.add(innerQuery, Occur.MUST);
+                addedFilters++;
             }
         }
+        if (addedFilters == 0) {
+            // no inner lucene query exists, ignore upstream
+            return null;
```

Comment:
The parser so far returned null when there where no inner clauses (lucene queries). We need to do the same if none of our new builder returns a query.